libcaf_core/caf/scheduled_actor.cpp

```diff
@@ -359,13 +371,27 @@ void scheduled_actor::quit(error x) {
   set_error_handler(silently_ignore<error>);
   // Drop future messages and produce sec::request_receiver_down for requests.
   set_default_handler(drop_after_quit);
-  // Cancel any active flow.
+  // Cancel flows and streams.
+  stream_sources_.clear();
+  if (!stream_subs_.empty()) {
+    decltype(stream_subs_) subs;
+    subs.swap(stream_subs_);
+    for (auto& [id, ptr] : subs)
+      ptr->cancel();
+  }
+  if (!stream_bridges_.empty()) {
+    decltype(stream_bridges_) bridges;
+    bridges.swap(stream_bridges_);
+    for (auto& [id, ptr] : bridges)
+      ptr->drop();
+  }
   while (!watched_disposables_.empty()) {
-    for (auto& ptr : watched_disposables_)
+    decltype(watched_disposables_) disposables;
+    disposables.swap(watched_disposables_);
+    for (auto& ptr : disposables)
       ptr.dispose();
-    watched_disposables_.clear();
-    run_actions();
   }
+  run_actions();
```

Comment:
Since this is repeated, it would look good factored out into its own function, IMHO.